app/src/main/java/com/asfoundation/wallet/ui/iab/LocalPaymentInteractor.kt

```diff
@@ -55,10 +55,10 @@ class LocalPaymentInteractor(private val deepLinkRepository: InAppDeepLinkReposi
     return (status == PENDING_USER_PAYMENT && type == "TOPUP") || (status == COMPLETED && (type == "INAPP" || type == "INAPP_UNMANAGED")) || status == FAILED || status == CANCELED || status == INVALID_TRANSACTION
   }
 
-  fun getCompletePurchaseBundle(isInApp: Boolean, merchantName: String, sku: String?,
+  fun getCompletePurchaseBundle(type: String, merchantName: String, sku: String?,
                                 scheduler: Scheduler,
                                 orderReference: String?, hash: String?): Single<Bundle> {
-    return if (isInApp && sku != null) {
+    return if (isInApp(type) && sku != null) {
       billing.getSkuPurchase(merchantName, sku, scheduler)
           .map {
             billingMessagesMapper.mapPurchase(it,
```

Comment:
The code inside the map function could be simplified to a single line

Reply:
done